### PR TITLE
add ticket type setting to target  ticket

### PR DESCRIPTION
**Pull request by btry**
### Changes description

Remove the relation between a target ticket and a ticket template.

Add a ticket type for request / incident

Migrate ticket type and predefined actors to the target ticket

### Checklist

Please check if your PR fulfills the following specifications:

- [ ] Tests for the changes have been added
- [ ] Docs have been added/updated

### References

Closes #N/A

### CODE REVIEW

File: inc/targetticket.class.php [hunk 1]
```diff
@@ -35,6 +35,17 @@
 
 class PluginFormcreatorTargetTicket extends PluginFormcreatorTargetBase
 {
+
+   const TICKET_TYPE_INCIDENT = 1;
+   const TICKET_TYPE_REQUEST = 2;
+
+   public static function getEnumTicketType() {
+      return [
+         PluginFormcreatorTargetTicket::TICKET_TYPE_INCIDENT  => __('Incident', 'formcreator'),
```
Comment (orthagh) [inline inc/targetticket.class.php]: Why not reusing the constants of Ticket class ?

**Comment by orthagh** [comment]
BTW, ticket's template also give predefined ticket's tasks so remove the relation is not a good idea

**Comment by btry** [comment]
> BTW, ticket's template also give predefined ticket's tasks so remove the relation is not a good idea.
Tickets's documents also

Only the type of ticket and actors are retrieved by the plugin. Only the ticket type is generally used because the actors are handled by the plugin. Removing the support of template helps to enhance the UX by configuring all settings from the plugin. Unfortunately the ticket templates don't allow everything permitted by the plugin (example : link tickets, add several actors of the same type).

File: inc/targetticket.class.php (hunk 1 above)
Comment (btry) [reply inc/targetticket.class.php]: changed

**Comment by orthagh** [comment]
I insist, we currently have customers using the **task**'s template by a **ticket**'s template predefined in form. So no, `type` is not the only remaining field set by formcreator.

Also, linked documents and hidden fields on final ticket

**Comment by btry** [comment]
OK, then I'll change the PR to provide the ticket type without need of any template, and keep the template association.

File: inc/targetticket.class.php [hunk 2]
```diff
@@ -541,6 +573,38 @@ public function pre_deleteItem() {
...
+         unset($predefined_fields['_groups_id_observer']);
+      }
+      if (isset($predefined_fields['_groups_id_assign'])) {
+         $this->addGroupActor('assigned', $predefined_fields['_groups_id_assign']);
+         unset($predefined_fields['_groups_id_assign']);
+      }
+      $data = array_merge($data, $predefined_fields);
+
+   }
+
```
Comment (btry) [inline inc/targetticket.class.php]: moved from save()

**Pull request merged**